ClientApp/src/app/Services/user.service.ts

```diff
@@ -24,27 +26,58 @@ export class UserService {
   constructor(private http: HttpClient) {
   }
 
-  register(model: RegisterModel): Observable<ResponseModel> {
+  registerUser(model: RegisterModel): Observable<ResponseModel> {
     return this.http.post<ResponseModel>(this.usersUrl + '/register', model, this.httpOptions).pipe(
       catchError(error => this.handleError(error))
     );
   }
 
-  logIn(model): Observable<ResponseModel>{
+  getUser(): Observable<User> {
+    const id = this.getUserId();
+    const url = `${this.usersUrl}/${id}`;
+    return this.http.get<User>(url);
+  }
+
+  logIn(model): Observable<ResponseModel> {
     return this.http.post<ResponseModel>(this.usersUrl + '/authenticate', model, this.httpOptions).pipe(
       catchError(error => this.handleError(error))
     );
   }
 
+  updateUser(user: UpdateUserModel): Observable<ResponseModel> {
+    return this.http.put<ResponseModel>(this.usersUrl + '/update', user, this.httpOptions).pipe(
+      catchError(error => this.handleError(error))
+    );
+  }
+
+  changePassword(password): Observable<ResponseModel> {
+    return this.http.put<ResponseModel>(this.usersUrl + '/update-password', password, this.httpOptions).pipe(
+      catchError(error => this.handleError(error))
+    );
+  }
+
   get isLoggedIn(): boolean {
     return localStorage.getItem('token') !== null;
   }
 
-  logout(): void {
+  logOut(): void {
     localStorage.removeItem('token');
   }
 
-  handleError(error: HttpErrorResponse): Observable<any>{
+  deleteUser(id: number): Observable<ResponseModel> {
+    const url = `${this.usersUrl}/${id}`;
+    return this.http.delete<ResponseModel>(url, this.httpOptions).pipe(
+      catchError(error => this.handleError(error)));
```

Comment:
Please place ); on a new line as you did in other pipe() calls above, be consistent with your code style.